hibernate-core/src/main/java/org/hibernate/loader/BatchLoadSizingStrategy.java

```diff
@@ -0,0 +1,16 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.loader;
+
+/**
+ * Strategy (pluggable) for determining an optimal size for batch loads.
+ *
+ * @author Steve Ebersole
+ */
+public interface BatchLoadSizingStrategy {
+	int determineOptimalBatchLoadSize(int numberOfKeyColumns, int numberOfKeys);
```

Comment:
>  Do you typically fill out JavaDoc for such prototyping?

Yes, I do it where I think it helps reviewers to better understand my draft.

> Well numberOfKeyColumns is important to be able...

Got it, thanks.

> So if you have something you'd like to see different that is the point of this. But really constructive criticism is usually better. So maybe rather than just pointing out what you don't like you could propose an alternative?

Where did I express any dislike? I asked one *question* for understanding, followed by a *proposal* how others could be prevented from asking the same question. I find that very constructive.

Then I made another *suggestion* (in form of a question). Granted, that suggestion was based on a wrong understanding of the API's purpose - which I acknowledged in my second comment.

I don't see how this can be conceived as not constructive, I am sorry if you got that impression. 

Anyways, the contract looks good in this light. Introducing a parameter object may be a good idea with respect to evolution of the API (hint: this is a proposal for an alternative ;).